scripts/eurostat/birth_death_migration/import_data.py

```diff
@@ -0,0 +1,131 @@
+
+        preprocessed_df = (preprocessed_df.set_index(["geo", "time"]).pivot(
+            columns="statistical_variable")['value'].reset_index().rename_axis(
+                None, axis=1))
+    # Fill missing 'geo' values with a colon.
+    preprocessed_df.fillna(': ', inplace=True)
+
+    # Split notes out of values.
+    # Ex: "5598920 b" -> "5598920", "b".
+    # Ex: "5665118" -> "5665118", ""
+    # Ex: ": c" -> "", "c"
+    # Append extra space for all values that do not come with a note.
+    preprocessed_df[preprocessed_df.columns[2:]] = preprocessed_df[
+        preprocessed_df.columns[2:]].replace("([0-9:])$",
+                                             lambda m: m.group(0) + ' ')
+    # replace comma in column names with a vertical bar so that we can save it in csv later
+    new_column_names = []
+    for column_name in preprocessed_df.columns:
+        new_column_names.append(column_name.replace(',', '|'))
+    preprocessed_df.columns = new_column_names
+    for column_name in preprocessed_df.columns[2:]:
+        preprocessed_df['{0}'.format(column_name)], preprocessed_df[
+            '{0}|notes'.format(column_name)] = (zip(
+                *preprocessed_df[column_name].apply(lambda x: x.split(' '))))
+    return preprocessed_df
+
+
+def clean_data(preprocessed_df, output_path):
+    """Drops unnecessary columns that are not needed for data import and reformat column names."""
+    if preprocessed_df is None:
+        raise ValueError("Uninitialized value of processed data frame. "
+                         "Please check you are calling preprocess_data "
+                         "before clean_data.")
+    # drop unused columns
+    clean_df = preprocessed_df[preprocessed_df.columns[:len(
+        preprocessed_df.columns
+    ) // 2 + 1]]  # number of columns should be 2 + 2X, we want the first 2 + X
+    # columns
+    # replace colon with NaN.
+    clean_df = clean_df.replace(':', '')
+    clean_df.to_csv(output_path, index=False)
+    clean_df = pd.read_csv(output_path)
+
+    clean_df['geo'] = 'dcid:nuts/' + clean_df['geo']
+    original_names = [
+        'geo', 'time', 'DEATH', 'GBIRTHRT', 'GDEATHRT', 'GROW', 'GROWRT', 'JAN',
+        'LBIRTH'
+    ]
+    new_names = [
+        'geo', 'time', 'Count_MortalityEvent',
+        'Count_BirthEvent_AsAFractionOfCount_Person',
+        'Count_MortalityEvent_AsAFractionOfCount_Person',
+        'IncrementalCount_Person', 'GrowthRate_Count_Person', 'Count_Person',
+        'Count_BirthEvent'
+    ]
+    clean_df = clean_df[original_names]
+    clean_df[['GBIRTHRT', 'GDEATHRT',
+              'GROWRT']] /= 1000  # apply scaling factor of 1000
+    clean_df.columns = new_names
+    clean_df.to_csv(output_path, index=False)
+
+
+if __name__ == '__main__':
+    DATA_LINK = "https://ec.europa.eu/eurostat/estat-navtree-portlet-prod/BulkDownloadListing?file=data/demo_r_gind3.tsv.gz"
+    FINAL_CSV_NAME = 'demo_r_gind3_final_stage.csv'
```

Comment:
Please move constants to the top of the file (under the imports). Also, add a _ if the constant is not used outside this file. You may revisit https://google.github.io/styleguide/pyguide.html when you have some time.

Reply:
decided to refactor these to be normal variable since it's only used once.